Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,656 @@
+/**
+ * @fileoverview
+ * This file contains the majority of the functionality used across the various web pages, in the following order
+ * 1: universal functions
+ * 2: functions related to invoking methods
+ * 3: camera related functions
+ */
+
+
+//immutable global variables are written in all caps
+const PORT = 5000;
+
+/**
+ * @global variables. Mutable global variables in javascript are typically named in camelCase
+ */
+var graphTopologies = {};
+var graphInstances = {};
+var cameras = {};
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// universal functions
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getCamerasFromServer()
+{
+    return new Promise((resolve, reject) =>
+    {
+        let request = prepareRequest(`http://localhost:${PORT}/globals`, "GET");
+        request.onreadystatechange = function ()
+        {
+            if (request.readyState == 4)
+            {
+                if (request.status == 200)
+                {
+                    let response = JSON.parse(request.response);
+                    cameras = response.cameras;
+                    resolve(response);
+                }
+                else
+                {
+                    reject("Get Globals rejection");
+                }
+            }
+            else if (request.status != 200)
+            {
+                reject("Server not responding");
+            }
+        }
+        request.send();
+    });
+}
+
+/**
+* send current values to the server to be stored for later 
+*/
+function sendCamerasToServer()
+{
+    let globals =
+    {
+        cameras: cameras
+    };
+    let request = prepareRequest(`http://localhost:${PORT}/globals`, "PUT");
+    request.send(JSON.stringify(globals));
+}
+
+/**
+*  generate request with given parameters to a defined url in server. default method is POST
+* @param {string} url - @link to communicate on
+* @param {string} requestType - POST or GET request 
+* @returns {XMLHttpRequest} - generated, unsent request
+*/
+function prepareRequest(url, requestType = "POST")
+{
+    let request = new XMLHttpRequest();
+    request.open(requestType, url, true);
+    // I would prefer the timeout to be shorter, for security purposes, however when we invoke LVA methods on the device the default max response time in seconds = 200
```

Comment:
two things: 1) is lva slow enough that you can't just turn down the timeout? i think you said yes? 2) if the lower layer timeout is 200 seconds, the upper layer needs to be a little higher or they completions can race. add a few seconds.

Reply:
It seems to be fine with less time. I've now set the LVA response time to 80 seconds, and the XMLHttpRequest timeout to 110 seconds